### PR TITLE
support SFSpeechRecognizer

### PR DESCRIPTION
**Pull request by dimohamdy**
support SFSpeechRecognizer for iOS 10

### CODE REVIEW

**Comment by dimohamdy** [comment]
don't forget to add  this in plist

```
    <key>NSMicrophoneUsageDescription</key>
    <string>Your microphone will be used to record your speech when you press the "Start Recording" button.</string>
    <key>NSSpeechRecognitionUsageDescription</key>
    <string>Speech recognition will be used to determine which words you speak into this device's microphone.</string>
```

**Comment by dimohamdy** [comment]
can i change the deployment target to iOS 10 to fix error
http://stackoverflow.com/questions/38536956/conditionally-import-a-framework-such-as-speech-based-on-ios-version-in-swift/38538861

File: Source/PermissionTypes/SpeechRecognizer.swift [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    }
+    
+    func requestSpeechRecognizer(callback: Callback) {
+
+        SFSpeechRecognizer.requestAuthorization { _ in
+                    callback(self.statusSpeechRecognizer)
+
+            }
+    }
+}
```
Comment (delba) [inline Source/PermissionTypes/SpeechRecognizer.swift:32]: Could you please adopt the project's coding style ? MIT License and number of spaces/indentation etc. ?
It seems like details but it greatly helps people who aren't familiar with the project to navigate quickly.

File: Source/Permission.swift [hunk 2]
```diff
@@ -186,6 +192,7 @@ public class Permission: NSObject {
         case .Events:            requestEvents(callback)
         case .Bluetooth:         requestBluetooth(self.callback)
         case .Motion:            requestMotion(self.callback)
+        case .SpeechRecognizer:   requestSpeechRecognizer(callback)
```
Comment (delba) [inline Source/Permission.swift:195]: Indentation

**Comment by delba** [comment]
Concerning the keys to add to your plist, check out what we did for `Location`:

``` swift
func requestLocationAlways(callback: Callback) {
        guard let _ = NSBundle.mainBundle().objectForInfoDictionaryKey(.nsLocationAlwaysUsageDescription) else {
            print("WARNING: \(.nsLocationAlwaysUsageDescription) not found in Info.plist")
            return
        }
       // ...
```
- https://github.com/delba/Permission/blob/master/Source/PermissionTypes/LocationAlways.swift#L42 
- https://github.com/delba/Permission/blob/master/Source/PermissionTypes/LocationWhenInUse.swift#L41

**Comment by delba** [comment]
Also we aren't using Xcode 8 on this branch. The future is [`swift-3.0`](https://github.com/delba/Permission/tree/swift-3.0)

**Pull request closed without merging**